### ISSUE TITLE
Reject over window function

### ISSUE THREAD

**Issue by marvenlee2486** [singularity-data/risingwave#3963]
**Describe the bug**
```
postgres=# select round(x) over (Partition by y) from t3;
ERROR:  OVER specified, but round is not a window function nor an aggregate function
LINE 1: select round(x) over (Partition by y) from t3;
```

in RW
```
dev=> select round(x) over (Partition by y) from t3;
 ?column? 
----------
(0 rows)
```

https://www.postgresql.org/docs/14/sql-expressions.html
**To Reproduce**
Steps to reproduce the behavior.

**Expected behavior**

**Additional context**

**Comment by xiangjinwu** [comment]
To clarify, `over (window function)` is not implemented yet. The short term fix is to report proper error.

**Comment by fuyufjh** [comment]
> To clarify, over (window function) is not implemented yet. The short term fix is to report unimplemented error, no matter it is agg call or non-agg function.

Is it done?

**Comment by xiangjinwu** [comment]
Fixed by #4073
